airflow/models/baseoperator.py

```diff
@@ -1325,7 +1335,7 @@ def run(
                     execution_date=info.logical_date,
                     data_interval=info.data_interval,
                 )
-                ti = TaskInstance(self, run_id=None)
+                ti = TaskInstance(self, run_id=dr.run_id)
```

Comment:
Ditto. (separation out ? or changing commit subject).

Reply:
Oh I remember this one. This is actually a case of "it makes no difference at run time" since `ti.dag_run = dr` on the next line has the same end effect, but doing it this way makes mypy happier.